tests/test_event_processor.py

```diff
@@ -0,0 +1,290 @@
+    event_dispatcher = TestEventDispatcher()
+
+    with mock.patch.object(self.optimizely, 'logger') as mock_config_logging:
+      self._set_event_processor(event_dispatcher, mock_config_logging)
+
+    self.project_config.revision = 1
+    self.project_config.project_id = 'X'
+
+    user_event_1 = self._build_conversion_event(self.event_name, self.project_config)
+    self._event_processor.process(user_event_1)
+    mock_config_logging.debug.assert_called_with('Received user_event: ' + str(user_event_1))
+    event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+    self.project_config.revision = 2
+    self.project_config.project_id = 'X'
+
+    user_event_2 = self._build_conversion_event(self.event_name, self.project_config)
+    self._event_processor.process(user_event_2)
+    mock_config_logging.debug.assert_called_with('Received user_event: ' + str(user_event_2))
+    event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+    time.sleep(1.5)
+
+    self.assertStrictTrue(event_dispatcher.compare_events())
+    self.assertEqual(0, self._event_processor.event_queue.qsize())
+
+  def test_flush_on_mismatch_project_id(self):
+    event_dispatcher = TestEventDispatcher()
+
+    with mock.patch.object(self.optimizely, 'logger') as mock_config_logging:
+      self._set_event_processor(event_dispatcher, mock_config_logging)
+
+    self.project_config.revision = 1
+    self.project_config.project_id = 'X'
+
+    user_event_1 = self._build_conversion_event(self.event_name, self.project_config)
+    self._event_processor.process(user_event_1)
+    mock_config_logging.debug.assert_called_with('Received user_event: ' + str(user_event_1))
+    event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+    self.project_config.revision = 1
+    self.project_config.project_id = 'Y'
+
+    user_event_2 = self._build_conversion_event(self.event_name, self.project_config)
+    self._event_processor.process(user_event_2)
+    mock_config_logging.debug.assert_called_with('Received user_event: ' + str(user_event_2))
+    event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+    time.sleep(1.5)
+
+    self.assertStrictTrue(event_dispatcher.compare_events())
+    self.assertEqual(0, self._event_processor.event_queue.qsize())
+
+  def test_stop_and_start(self):
+    event_dispatcher = TestEventDispatcher()
+
+    with mock.patch.object(self.optimizely, 'logger') as mock_config_logging:
+      self._set_event_processor(event_dispatcher, mock_config_logging)
+
+    user_event = self._build_conversion_event(self.event_name, self.project_config)
+    self._event_processor.process(user_event)
+    mock_config_logging.debug.assert_called_with('Received user_event: ' + str(user_event))
+    event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+    time.sleep(1.5)
+
+    self.assertStrictTrue(event_dispatcher.compare_events())
+    self._event_processor.close()
+
```

Comment:
check this "self.assertEqual(0, self._event_processor.event_queue.qsize())" here as well